File: taller/AMarotoChivite-taller/AMarotoChivite-taller-Java/src/main/java/simulacionTaller/models/Electrician.java
package simulacionTaller.models;

public class Electrician extends Worker {
    // Campo de clase protected para mantener el incremento, también en Multi
    protected static int countElectrician = 0;

    private final int id = countElectrician;

    public Electrician(String name, int experienceYear, double salario, int dailyHours, BossAssigment bossAssigment) {
        super(name, experienceYear, salario, dailyHours, bossAssigment);
        autoCountElectrician();
    }

    /**
     * Creamos un get, para encapsular el valor del contador y no poderlo manipular
     */
    public static int getCountElectrician() {
        return countElectrician;
    }

    /**
     * El electricista arreglará la electricidad
     *
     * @return mensaje del electricista arreglando
     */
    public String fixElectrician() {
        return "El electricista ha arreglado la electricidad... ";
    }

    /**
     * El electricista está comiendo
     *
     * @return mensaje del comiendo
     */
    public String eatAsElectrician() {
        return "El electricista está comiendo... ";
    }

    /**
     * Aumentamos un ID por cada jefe que hemos creado, para disponer de un contador
     */
    private void autoCountElectrician() {
        countElectrician++;
    }

    @Override
    public String toString() {
        return "Trabajador " + (id + 1) + " Electricista " + (id + 1) + "{" +
                " name= " + name +
                ", salario=" + salario +
                ", horas/día trabajadas= " + dailyHours +
                ", annoExperiencia=" + experienceYear +
                '\'' +
                "}";
    }
}

File: taller/AMarotoChivite-taller/AMarotoChivite-taller-kotlin/src/main/kotlin/simulacionTaller/models/Electrician.kt
package simulacionTaller.models

class Electrician(name: String?, experienceYear: Int, salario: Double, dailyHours: Int, bossAssigment: BossAssigment) :
    Worker(name, experienceYear, salario, dailyHours, bossAssigment) {
    private val idElectrician = countElectrician

    init {
        autoCountElectrician()
    }

    /**
     * El electricista arreglará la electricidad
     *
     * @return mensaje del electricista arreglando
     */
    fun fixElectrician(): String {
        return "El electricista ha arreglado la electricidad... "
    }

    /**
     * El electricista está comiendo
     *
     * @return mensaje del comiendo
     */
    fun eatAsElectrician(): String {
        return "El electricista está comiendo... "
    }

    /**
     * Aumentamos un ID por cada jefe que hemos creado, para disponer de un contador
     */
    private fun autoCountElectrician() {
        countElectrician++
    }

    override fun toString(): String {
        return "Trabajador " + (idElectrician + 1) + " Electricista " + (idElectrician + 1) + "{" +
                " name= " + name +
                ", salario=" + salario +
                ", horas/día trabajadas= " + dailyHours +
                ", annoExperiencia=" + experienceYear +
                '\'' +
                "}"
    }

    companion object {
        /**
         * Creamos un get, para encapsular el valor del contador y no poderlo manipular
         */
        // Campo de clase protected para mantener el incremento, también en Multi
        var countElectrician = 0
    }
}